src/Core/Services/GraphQLSchemaCreator.cs

```diff
@@ -251,6 +266,237 @@ private DocumentNode GenerateSqlGraphQLObjects(RuntimeEntities entities, Diction
             return new DocumentNode(nodes);
         }
 
+        /// <summary>
+        /// Helper method to traverse through all the relationships for all the entities exposed in the config.
+        /// For all the relationships defined in each entity's configuration, it adds an referencing field directive to all the
+        /// referencing fields of the referencing entity in the relationship. For relationships defined in config:
+        /// 1. If an FK constraint exists between the entities - the referencing field directive
+        /// is added to the referencing fields from the referencing entity.
+        /// 2. If no FK constraint exists between the entities - the referencing field directive
+        /// is added to the source.fields/target.fields from both the source and target entities.
+        ///
+        /// The values of such fields holding foreign key references can come via insertions in the related entity.
+        /// By adding ForiegnKeyDirective here, we can later ensure that while creating input type for create mutations,
+        /// these fields can be marked as nullable/optional.
+        /// </summary>
+        /// <param name="objectTypes">Collection of object types.</param>
+        /// <param name="entities">Entities from runtime config.</param>
+        private void AddReferencingFieldDirective(RuntimeEntities entities, Dictionary<string, ObjectTypeDefinitionNode> objectTypes)
+        {
+            foreach ((string sourceEntityName, ObjectTypeDefinitionNode sourceObjectTypeDefinitionNode) in objectTypes)
+            {
+                Entity entity = entities[sourceEntityName];
+                if (!entity.GraphQL.Enabled || entity.Relationships is null)
+                {
+                    continue;
+                }
+
+                string dataSourceName = _runtimeConfigProvider.GetConfig().GetDataSourceNameFromEntityName(sourceEntityName);
+                ISqlMetadataProvider sqlMetadataProvider = _metadataProviderFactory.GetMetadataProvider(dataSourceName);
+                SourceDefinition sourceDefinition = sqlMetadataProvider.GetSourceDefinition(sourceEntityName);
+                Dictionary<string, FieldDefinitionNode> sourceFieldDefinitions = sourceObjectTypeDefinitionNode.Fields.ToDictionary(field => field.Name.Value, field => field);
+
+                // Retrieve all the relationship information for the source entity which is backed by this table definition.
```

Comment:
Why would we need to fail? Its just that nested insertion is supported for tables only as of now. And nested insertions are enabled globally, not on entity level. The user just won't see the schema generated to do nested insertion in database views.
I will add an additional check wherever required to ensure we are doing everything related to nested insertion only for tables.

Reply:
Im wondering if `relationship` section in config is only allowed for tables. i.e. if the views/stored proc have relationship section, dab validate should detect it and fail in validation phase since I dont think we allow relationships between stored procs even for queries. Im not sure about views though. if we support relationship between views for querying, we should add backlog to support nested inserts for views at the very least.

Reply:
No, we don't support defining relationships for non-table entities:

https://github.com/Azure/data-api-builder/blob/9e377c374b1d175725cd0556379ad33d2522b46a/src/Core/Configurations/RuntimeConfigValidator.cs#L801